### DISCUSSION TITLE
Possibility to install WebSphere as a non-root user

### DISCUSSION THREAD

**Discussion by martdj** [HCL-TECH-SOFTWARE/connections-automation#113] [Ideas]
For security reasons, many companies prefer to not install WebSphere and IHS as the root user. If I try to run the playbook with remote_user: wasusr and all necessary directories are owned by this user, I still get an error that the sudo password is not provided, even though it's not needed. I guess this is currently not working. Is this a scenario that you would be willing to look into?

**Comment by dejo1307** [comment]
I don't mind adding it. Do you know, by any chance, any documentation where the process is described? I just tried to quickly google it but there's lot of documentation out there and nothing really straight forward. If not it's also fine, will figure it out.

**Reply by martdj** [reply]
I don't know of any definitive manual on how to do a Connections installation where WebSphere is installed with a non-root user, but it's not that much different from a normal installation. IBM Installation Manager needs to be installed with userinstc instead of installc and all installations with Installation manager will need to be started with the non-root user.
Personally, I use wasusr as non-root user and install in:
/appl/was for WebSphere
/appl/ihs for IBM HTTP Server
/appl/ihsplugins for the WebSphere plugins
This all because the root-owned /opt doesn't make the most sense to install the software in. You also need to take into account that IHS can't use ports below 1024, so 80 and 443 are out.
